### PR TITLE
feat(subscribers): opt to not create subscriptions

### PR DESCRIPTION
**Pull request by riezebosch**
fix #94 
---
Rebus is [MIT-licensed](https://opensource.org/licenses/MIT). The code submitted in this pull request needs to carry the MIT license too. By leaving this text in, __I hereby acknowledge that the code submitted in the pull request has the MIT license and can be merged with the Rebus codebase__.

### CODE REVIEW

**Comment by ecw1973** [comment]
just discovered this PR when I was about to create one.   I just made similar changes as I need this feature as well because our service bus is enterprise wide and is treated as a managed resource where creation of new topics/queues/subscriptions is done via IaC and requires going through an approval process.

File: Rebus.AzureServiceBus/AzureServiceBus/AzureServiceBusTransport.cs [hunk 1]
```diff
@@ -146,6 +146,7 @@ public async Task<string[]> GetSubscriberAddresses(string topic)
     /// </summary>
     public async Task RegisterSubscriber(string topic, string subscriberAddress)
     {
+        if (DoNotCreateSubscriptionsEnabled) return;
```
Comment (ecw1973) [inline Rebus.AzureServiceBus/AzureServiceBus/AzureServiceBusTransport.cs:149]: I suggest adding a log message in order to be consistent with the other DoNot properties implementations.

File: Rebus.AzureServiceBus/Config/AzureServiceBusTransportSettings.cs [hunk 2]
```diff
@@ -177,4 +178,14 @@ public AzureServiceBusTransportSettings SetReceiveOperationTimeout(TimeSpan rece
         ReceiveOperationTimeout = receiveOperationTimeout;
         return this;
     }
+
```
[the remark this answers is not in the document]
Comment (riezebosch) [reply Rebus.AzureServiceBus/Config/AzureServiceBusTransportSettings.cs:181]: Would even go that far that the management client is injected into the transport and you have the option to supply/decorate a client that doesn't do anything.

**Comment by ecw1973** [comment]
FYI, I discovered Rebus works as-is (i.e. using DoNotCreateQueues().DoNotCheckQueueConfiguration()) as long as you do not make any Subscribe calls.   I was using common code for multiple service bus namespaces (some of which do not allow manage) and this was my issue.

**Comment by mookid8000** [comment]
What does this PR actually do? You can easily avoid creating/interrogating subscriptions by avoiding making any calls to

```csharp
await bus.Subscribe<TEvent>();
// and
await bus.Unsubscribe<TEvent>();
```

as well as their generic siblings

```csharp
var topics = bus.Advanced.Topics;

await topics.Subscribe("some-topic");
// and
await topics.Unsubscribe("some-topic");
```

**Comment by riezebosch** [comment]
but do I then still receive any messages?

**Pull request closed without merging**